src/promptflow/promptflow/executor/_line_execution_process_pool.py

```diff
@@ -208,65 +180,131 @@ def __exit__(self, exc_type, exc_val, exc_tb):
+            else:
+                # Handle line execution completed.
+                self._completed_idx[line_number] = format_current_process(
+                    process_name, process_id, line_number, is_completed=True)
 
             self._processing_idx.pop(line_number)
-            log_progress(
-                run_start_time=run_start_time,
-                logger=bulk_logger,
-                count=len(result_list),
-                total_count=self._nlines,
-            )
+
+    def _get_process_info(self, index, input_queue, output_queue):
+        while True:
+            try:
+                process_id = self._process_info[index].process_id
+                process_name = self._process_info[index].process_name
+                process_info = ProcessInfo(index=index, process_id=process_id, process_name=process_name,
+                                           input_queue=input_queue, output_queue=output_queue)
+                return process_info
+            except KeyError:
+                continue
+            except Exception as e:
+                bulk_logger.info(
+                    f"Unable to access shared dictionary 'process_info', possibly "
+                    f"because the sub process is down. Exception: {e}")
+
+    def _process_output_message(self, message, result_list):
+        if isinstance(message, LineResult):
+            message = self._process_multimedia(message)
+            result_list.append(message)
+            return True
+        elif isinstance(message, FlowRunInfo):
+            self._storage.persist_flow_run(message)
+        elif isinstance(message, NodeRunInfo):
+            self._storage.persist_node_run(message)
+
+        return False
+
+    def handle_line_timeout(self, line_number, timeout_time, inputs, run_id, start_time, result_list):
+        bulk_logger.warning(f"Line {line_number} timeout after {timeout_time} seconds.")
+        ex = LineExecutionTimeoutError(line_number, timeout_time)
+        result = self._generate_line_result_for_exception(
+            inputs, run_id, line_number, self._flow_id, start_time, ex
+        )
+        result_list.append(result)
+
+    def handle_process_crashed(self, line_number, inputs, run_id, start_time, result_list):
+        bulk_logger.warning(f"Process crashed while executing line {line_number},")
+        ex = ProcessCrashError(line_number)
+        result = self._generate_line_result_for_exception(
+            inputs, run_id, line_number, self._flow_id, start_time, ex
+        )
+        result_list.append(result)
+
+    def _monitor_process_and_manage_tasks_and_results(
+            self,
+            task_queue: Queue,
+            timeout_time,
+            result_list,
+            index,
+            input_queue,
+            output_queue
+    ):
+        process_info = self._get_process_info(index, input_queue, output_queue)
+
+        self._process_task(
```

Comment:
Seems this `_process_task` is actually `monitor_workers_and_process_tasks_in_thread`?

Reply:
Yes, remove '_process_task' and move it in 'monitor_workers_and_process_tasks_in_thread'